test/puppetlabs/jdbc_util/core_test.clj

```diff
@@ -448,3 +470,56 @@
     (testing "when there is no associated sequence, reconcile-sequence-for-column! throws an exception"
       (is (thrown-with-msg? Exception #"No sequence found"
                             (reconcile-sequence-for-column! test-db "authors" "name"))))))
+                            
+(deftest obj->jsonb-test
+  (testing "objects are converted to jsonb format successfully"
+    (let [obj {:foo "bar" :baz "qux"}
+          string "foobar"
+          num 1234
+          jsonb-string (obj->jsonb string)
+          jsonb-obj (obj->jsonb obj)
+          jsonb-num (obj->jsonb num)]
+      (is (= (type jsonb-string) org.postgresql.util.PGobject))
+      (is (= (type jsonb-obj) org.postgresql.util.PGobject))
+      (is (= (type jsonb-num) org.postgresql.util.PGobject)))))
+
+(deftest parse-jsonb-object-test
+  (testing "if the resource is not a PGobject, the original object is returned"
+    (let [obj (parse-jsonb-object {:foo "bar"})
+          num (parse-jsonb-object 1234)
+          string (parse-jsonb-object "hello")]
+      (is (= true (map? obj)))
+      (is (= true (integer? num)))
+      (is (= true (string? string)))))
+  (testing "all PGobject values are returned, and jsonb values are returned parsed"
+    (let [pgtext (doto (PGobject.)
+                    (.setType "text")
+                    (.setValue "hello"))
+          text-result (parse-jsonb-object pgtext)
+          pgjson (doto (PGobject.)
+                    (.setType "json")
+                    (.setValue (json/generate-string {:foo "bar"})))
+          json-result (parse-jsonb-object pgjson)
+          pgjsonb (doto (PGobject.)
+                    (.setType "jsonb")
+                    (.setValue (json/generate-string {:yee "haw"})))
+          jsonb-result (parse-jsonb-object pgjsonb)]
+      (is (= true (string? text-result)))
+      (is (= text-result "hello"))
+      (is (= true (string? json-result)))
+      (is (= json-result "{\"foo\":\"bar\"}"))
+      (is (= true (map? jsonb-result)))
+      (is (= jsonb-result {:yee "haw"})))))
+      
+(deftest jsonb-converter-test
+  (testing "the converter function only returns parsed jsonb columns from the db"
+    (let [query (str "select * from jsonb_testing where text_stuff = 'text'")
+          convert-all-columns {:row-fn (jsonb-converter :name :text_stuff :json_stuff :jsonb_stuff)} 
+          results (jdbc/query test-db [query] convert-all-columns)]
+      (doseq [result results]
+        (is (= true (string? (:name result))))
+        (is (= true (string? (:text_stuff result))))
+        (is (= true (string? (:json_stuff result))))
+        (is (= "{\"should\":\"return encoded\"}" (:json_stuff result)))
+        (is (= true (map? (:jsonb_stuff result))))
```

Comment:
This could be more simply:
```(is (map? (:jsonb_stuff result))```

Reply:
Same with things above that compare against "true"

Reply:
Ah good point! Updating.